pennylane/devices/experimental/execution_config.py

```diff
@@ -68,6 +69,9 @@ class ExecutionConfig:
     derivative_order: int = 1
     """The derivative order to compute while evaluating a gradient"""
 
+    max_workers: Optional[int] = None
+    """Maximal number of processes for batch execution"""
+
```

Comment:
Eventually this will fall under `device_options`. The idea is that every device will care about things like `derivative_order` and `use_device_gradient`, but `device_options` will vary from device to device.

Reply:
Eventually, should not all devices support `max_workers>1`?

Reply:
I don't think it necessarily makes sense to add it to all devices.  Some, we don't want to actually put that much effort into them because they aren't that important. Some are already remote devices with parallelism (AWS).

Maybe if we do add it to all devices, we could add it as its own entry to `ExecutionConfig`?

Reply:
No problem, I'll move `max_workers` into `device_options`.